linetools/spectra/lsf.py

```diff
@@ -0,0 +1,295 @@
+        calibration products by intrument developers and should be loaded by 
+        self.load_XX_data() in the initialization stage of LSF(), where XX 
+        is the name of the instrument)
+
+        Parameters
+        ----------
+        wv0 : Quantity 
+            Wavelenght at which an LSF solution is required
+
+        Output
+        ------
+        LSF.lsf_wv0 : QTable
+            The interpolated lsf at wv0
+        """
+        #get wa0 to Angstroms
+        wv0 = wv0.to('AA')
+
+        #transform to wavelength in float() form assuming Angstroms
+        col_names = self._data.keys()
+        col_waves = [float(name.split('A')[0]) for name in col_names[1:]]
+                    
+        lsf_vals = []
+        for row in self._data:
+            aux_val = []
+            for i in range(1,len(row)):
+                aux_val += [row[i]]
+            f = interp1d(col_waves,aux_val,bounds_error=True,kind='linear') #we do not want to extrapolate
+            lsf_vals += [f(wv0.value)]
+        lsf_vals = np.array(lsf_vals)
+        #normalize
+        lsf_vals /= np.max(lsf_vals)
+
+        #create Column to store the interpolated LSF
+        #lsf_vals = Column(name='{:.0f}A'.format(wv0.value),data=lsf_vals)
+        lsf_vals = Column(name='kernel',data=lsf_vals)
+        
+        #create column of relative pixel in absolute wavelength
+        wv_array = [(self.pixel_scale * self._data['rel_pix'][i] + wv0).value for i in range(len(self._data))]
+        wv = Column(name='wv',data=wv_array, unit=u.AA)
+
+        #create lsf Table
+        lsf = Table()
+        lsf.add_column(wv)
+        lsf.add_column(lsf_vals)
+
+        #return lsf Table()
+        return lsf
+
+    def interpolate_to_wv_array(self,wv_array):
+        """
+        Given `wv_array` this function interpolates an LSF
+        to match both scale and extent of `wv_array`. Some checks 
+        are performed.
+
+        Parameters
+        ----------
+        wv_array : Quantity numpy.ndarray, shape(N,)
+            Wavelength array for which the LSF kernel is defined. The 
+            central wavelength value of `wv_array` define the wavelength
+            at which the LSF is defined, while the limits of `wv_array` 
+            define the extent of the kernel.
+
+
+        """
+        # Check correct format
+        if not ((isinstance(wv_array, np.ndarray)) or (isinstance(wv_array, Quantity))):
+            raise SyntaxError('`wv_array` must be Quantity numpy.ndarray')
+        elif len(wv_array.shape) != 1:
+            raise SyntaxError('`wv_array` must be of shape(N,), i.e. 1-dimensional array')
+        
+        #define useful quantities
+        wv_min = np.min(wv_array)
+        wv_max = np.max(wv_array)
+        wv0 = 0.5 * (wv_max + wv_min)
+        
+        lsf_tab = self.interpolate_to_wv0(wv0)
+        f = interp1d(lsf_tab['wv'],lsf_tab['kernel'],kind='cubic',bounds_error=False,fill_value=0)
```

Comment:
Akima looks good and robust to me for the LSFs; so I will do so. Just in case will allow two kinds of interpolation: ('cubic' and 'Akima'). I guess Akima interpolation should also be preferred for QSO continuum fitting (if not already implemented).